akka-discovery-azure-api/src/main/scala/akka/discovery/azureapi/Settings.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2017-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.discovery.azureapi
+
+import akka.actor.{
+  ActorSystem,
+  ClassicActorSystemProvider,
+  ExtendedActorSystem,
+  Extension,
+  ExtensionId,
+  ExtensionIdProvider
+}
+import com.typesafe.config.Config
+
+import java.util.Optional
+import scala.compat.java8.OptionConverters._
+
+final class Settings(system: ExtendedActorSystem) extends Extension {
```

Comment:
This is just used in one place, and is kept around as a val, so no need to have it be an extension, a regular class is enough.

Not super important, but, the usual pattern we use for settings is keep the settings class itself a regular case class with no knowledge of config or actor system and have a companion with `apply(Config)` and possibly `apply(ActorSystem)` for parsing/loading settings from config.

Also: `/** INTERNAL API */ @InternalApi`